src/compiler/parser.ts

```diff
@@ -2344,7 +2344,7 @@ namespace Parser {
                 return;
 
             case "is":
-                parseErrorAt(pos, scanner.getTextPos(), Diagnostics.A_type_predicate_is_only_allowed_in_return_type_position_for_functions_and_methods);
+                parseErrorAt(pos, scanner.getTokenEnd(), Diagnostics.A_type_predicate_is_only_allowed_in_return_type_position_for_functions_and_methods);

```

Comment:
this is also suspicious
Edit: No difference in tests when changing to getTokenStart, so I'm going with that.